Reference/Routing/Request-Pipeline/IContentFinder-v9.md

```diff
@@ -0,0 +1,221 @@
+        {
+            // Add our custom content finder just before the core ContentFinderByUrl
+            builder.ContentFinders().InsertBefore<ContentFinderByUrl, MyContentFinder>();
+            // You can also remove content finders, this is not required here though, since our finder runs before the url one
+            builder.ContentFinders().Remove<ContentFinderByUrl>();
+            // You use Append to add to the end of the collection
+            builder.ContentFinders().Append<AnotherContentFinderExample>();
+            // or Insert for a specific position in the collection
+            builder.ContentFinders().Insert<AndAnotherContentFinder>(3);
+            return builder;
+        }
+    }
+}
+```
+
+Then invoke it in `ConfigureServices` in the `Startup.cs` file:
+
+```c#
+public void ConfigureServices(IServiceCollection services)
+{
+#pragma warning disable IDE0022 // Use expression body for methods
+    services.AddUmbraco(_env, _config)
+        .AddBackOffice()
+        .AddWebsite()
+        .AddComposers()
+        .AddCustomContentFinders()
+        .Build();
+#pragma warning restore IDE0022 // Use expression body for methods
+}
+```
+
+
+#### Composer
+
+```csharp
+using Umbraco.Cms.Core.Composing;
+using Umbraco.Cms.Core.DependencyInjection;
+using Umbraco.Cms.Core.Routing;
+
+namespace RoutingDocs.ContentFinders
+{
+    public class UpdateContentFindersComposer : IUserComposer
+    {
+        public void Compose(IUmbracoBuilder builder)
+        {
+            // Add our custom content finder just before the core ContentFinderByUrl
+            builder.ContentFinders().InsertBefore<ContentFinderByUrl, MyContentFinder>();
+            // You can also remove content finders, this is not required here though, since our finder runs before the url one
+            builder.ContentFinders().Remove<ContentFinderByUrl>();
+            // You use Append to add to the end of the collection
+            builder.ContentFinders().Append<AnotherContentFinderExample>();
+            // or Insert for a specific position in the collection
+            builder.ContentFinders().Insert<AndAnotherContentFinder>(3);
+        }
+    }
+}
+
+```
+:::note
+In Umbraco7 there existed an IContentFinder that would find content and display it with an 'alternative template' via a convention. This could be to avoid the ugly `?alttemplate=blogfullstory` appearing on the querystring of the url when using the alternative template mechanism. Instead the Url could follow the convention of `/urltocontent/altemplatealias`. 
+
+Eg: `/blog/my-blog-post/blogfullstory` would 'find' the `/blog/my-blog-post` page and display using the `blogfullstory` template. 
+
+In Umbraco 8 this convention has been removed from the default configuration of Umbraco. You can reintroduce this behaviour by adding the `ContentFinderByUrlAndTemplate` ContentFinder back into the ContentFinderCollection, using an `IUserComposer`, or Umbraco builder extension (see above example).
+:::
```

Comment:
@nikolajlauridsen Im a little confused about this note here 🤔 You mention that it was something in V7 which have been removed in v8 but can be re-introduced, however this is a v9 article, is this also something that can be done in v9? If so then it might need to be made clear :)

Reply:
Ah yes, this is something that is removed by default in both V8 and V9, but you should be able to add it back in V9 as well, I'll double-check and then update the note 👍

Reply:
I've updated it so it says "In Umbraco 9 this convention has been removed", I don't know if there's a better way to explain it since it's the case for both v8 and v9.

Reply:
It looks fine and makes it clear that it have been removed in v9 as well :)